### PR TITLE
Support custom registries config

### PR DESCRIPTION
**Pull request by addyess**
Support setting up custom registries and authentication via a new option to configure the [`containerd-tempate.toml`](https://microk8s.io/docs/dockerhub-limits)

```bash
juju config microk8s custom_registries=\
  '[{"host": "docker.io", "url": "registry-1.docker.io", "username": "MYUSERNAME", "password": "MYPASSWORD"}]'
```

Without this, each unit will need to add the changes to the `containerd-template.toml` to add a custom registry or authentication for docker.io.  Use this to config to write those changes to containerd template for you.

### CODE REVIEW

File: config.yaml
Comment (stonepreston) [inline config.yaml:35]:
```suggestion
        e.g.: [{"url": "https://registry.example.com", "host": "my.registry:port", "username": "user", "password": "pass"}]
```

**Review by stonepreston — COMMENTED** [review]
small issue with required field missing in configuration comments, but this LGTM if its good with microk8s folks

File: tests/test_custom_registries.py
Comment (neoaggelos) [inline tests/test_custom_registries.py:125]: Should this be removed?

**Review by neoaggelos — COMMENTED** [review]
Nice work! Looks good but need one extra comment to be addressed

File: tests/test_custom_registries.py
Comment (addyess) [reply tests/test_custom_registries.py:125]: i.... i don't remember 😂 
I'll raise another commit here to figure out why i checked in a commented out test

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**
